subprojects/core/src/main/java/org/gradle/execution/TaskNameResolver.java

```diff
@@ -37,9 +39,12 @@ public class TaskNameResolver {
      * Non-exhaustively searches for at least one task with the given name, by not evaluating projects before searching.
      */
     public boolean tryFindUnqualifiedTaskCheaply(String name, ProjectInternal project) {
+        List<Project> allprojects = new ArrayList<>(project.getChildProjectsUnchecked().values());
+        allprojects.add(project);
+
         // don't evaluate children, see if we know it's without validating it
-        for (Project project1 : project.getAllprojects()) {
-            if (project1.getTasks().getNames().contains(name)) {
+        for (Project p : allprojects) {
+            if (p.getTasks().getNames().contains(name)) {
```

Comment:
❌  This seems to change the semantics. Previously, it was `allprojects`, meaning a full project sub-tree. Now, it's only the direct children of the given project (apart from the project itself).

If this change was not caught by an existing test, we should add a test that covers this case.

Reply:
Not actually, see [Project.getAllprojects](https://github.com/gradle/gradle/blob/master/subprojects/core-api/src/main/java/org/gradle/api/Project.java#L444)